frontend/controller/actions/group.js

```diff
@@ -250,7 +274,14 @@ export default (sbp('sbp/selectors/register', {
   'gi.actions/group/joinChatRoom': async function (params: GIActionParams) {
     try {
       const rootState = sbp('state/vuex/state')
-      const username = params.data.username || rootState.loggedIn.username
+      const rootGetters = sbp('state/vuex/getters')
+      const me = rootState.loggedIn.username
+      const username = params.data.username || me
+
+      if (!rootGetters.isJoinedChatRoom(params.data.chatRoomID) && username !== me) {
+        throw new GIErrorUIRuntimeError(L('Failed to join chat channel.'))
+      }
```

Comment:
What is this condition doing? It doesn't make sense to me.

Reply:
This is for blocking `not-chatroom-members` to add another to the chatroom.

Reply:
Then the error message should be a little clearer, maybe something like this:

```js
throw new GIErrorUIRuntimeError(L('Only channel members can invite others to join.'))
```

Reply:
Okay.